Description: The mesa t64 transition is not a thing on Bookworm, therefore
 we don't need this patch.
 .
 This reverts commit e06c71af05053440c9a576c5573865aef521ee25.
Author: Pascal Roeleven <dev@pascalroeleven.nl>

--- a/debian/changelog
+++ b/debian/changelog
@@ -101,11 +101,7 @@
   * Don't rely on module state in teedataobject_clear (from Brandt Bucher in
     https://github.com/python/cpython/issues/115874). Closes: #1063345.
 
-  [ Steve Langasek ]
-  * Rename libraries for 64-bit time_t transition, declare X-Time64-Compat
-    to get proper Provides (Steve Langasek). Closes: #1064336.
-
- -- Matthias Klose <doko@debian.org>  Thu, 29 Feb 2024 02:56:09 +0100
+ -- Colin Watson <cjwatson@debian.org>  Tue, 27 Feb 2024 10:15:37 +0000
 
 python3.12 (3.12.2-1) unstable; urgency=medium
 
--- a/debian/control
+++ b/debian/control
@@ -118,10 +118,7 @@
  This package contains some essential modules. It is normally not
  used on it's own, but as a dependency of python3.12-minimal.
 
-Package: libpython3.12t64
-Provides: ${t64:Provides}
-Replaces: libpython3.12
-Breaks: libpython3.12 (<< ${source:Version})
+Package: libpython3.12
 Architecture: any
 Multi-Arch: same
 Section: libs
@@ -148,7 +145,7 @@
 Multi-Arch: allowed
 Depends: python3.12 (= ${binary:Version}),
   libpython3.12-dev (= ${binary:Version}),
-  libpython3.12t64 (= ${binary:Version}),
+  libpython3.12 (= ${binary:Version}),
   ${shlibs:Depends}, ${misc:Depends}
 Recommends: libc6-dev | libc-dev
 Description: Header files and a static library for Python (v3.12)
@@ -164,7 +161,7 @@
 Multi-Arch: same
 Pre-Depends: ${misc:Pre-Depends}
 Depends: libpython3.12-stdlib (= ${binary:Version}),
-  libpython3.12t64 (= ${binary:Version}),
+  libpython3.12 (= ${binary:Version}),
   libexpat1-dev, zlib1g-dev,
   ${shlibs:Depends}, ${misc:Depends}
 Recommends: libc6-dev | libc-dev
@@ -226,7 +223,7 @@
 Section: debug
 Architecture: any
 Multi-Arch: allowed
-Depends: python3.12 (= ${binary:Version}), libpython3.12t64-dbg (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
+Depends: python3.12 (= ${binary:Version}), libpython3.12-dbg (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
 Recommends: gdb
 Suggests: python3-gdbm-dbg, python3-tk-dbg
 Description: Debug Build of the Python Interpreter (version 3.12)
@@ -239,9 +236,7 @@
  .
  See the README.debug for more information.
 
-Package: libpython3.12t64-dbg
-X-Time64-Compat: libpython3.12-dbg
-Provides: ${t64:Provides}
+Package: libpython3.12-dbg
 Section: debug
 Architecture: any
 Multi-Arch: same
--- a/debian/control.in
+++ b/debian/control.in
@@ -118,10 +118,7 @@
  This package contains some essential modules. It is normally not
  used on it's own, but as a dependency of @PVER@-minimal.
 
-Package: lib@PVER@t64
-Provides: ${t64:Provides}
-Replaces: lib@PVER@
-Breaks: lib@PVER@ (<< ${source:Version})
+Package: lib@PVER@
 Architecture: any
 Multi-Arch: same
 Section: libs
@@ -148,7 +145,7 @@
 Multi-Arch: allowed
 Depends: @PVER@ (= ${binary:Version}),
   lib@PVER@-dev (= ${binary:Version}),
-  lib@PVER@t64 (= ${binary:Version}),
+  lib@PVER@ (= ${binary:Version}),
   ${shlibs:Depends}, ${misc:Depends}
 Recommends: libc6-dev | libc-dev
 Description: Header files and a static library for Python (v@VER@)
@@ -164,7 +161,7 @@
 Multi-Arch: same
 Pre-Depends: ${misc:Pre-Depends}
 Depends: lib@PVER@-stdlib (= ${binary:Version}),
-  lib@PVER@t64 (= ${binary:Version}),
+  lib@PVER@ (= ${binary:Version}),
   libexpat1-dev, zlib1g-dev,
   ${shlibs:Depends}, ${misc:Depends}
 Recommends: libc6-dev | libc-dev
@@ -226,7 +223,7 @@
 Section: debug
 Architecture: any
 Multi-Arch: allowed
-Depends: @PVER@ (= ${binary:Version}), lib@PVER@t64-dbg (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
+Depends: @PVER@ (= ${binary:Version}), lib@PVER@-dbg (= ${binary:Version}), ${shlibs:Depends}, ${misc:Depends}
 Recommends: gdb
 Suggests: python3-gdbm-dbg, python3-tk-dbg
 Description: Debug Build of the Python Interpreter (version @VER@)
@@ -239,9 +236,7 @@
  .
  See the README.debug for more information.
 
-Package: lib@PVER@t64-dbg
-X-Time64-Compat: lib@PVER@-dbg
-Provides: ${t64:Provides}
+Package: lib@PVER@-dbg
 Section: debug
 Architecture: any
 Multi-Arch: same
--- /dev/null
+++ b/debian/libPVER-dbg.overrides.in
@@ -0,0 +1,20 @@
+lib@PVER@-dbg binary: package-name-doesnt-match-sonames
+lib@PVER@-dbg binary: non-dev-pkg-with-shlib-symlink
+
+# no, it's not unusual
+lib@PVER@-dbg binary: unusual-interpreter
+
+# just the gdb debug file
+lib@PVER@-dbg binary: python3-script-but-no-python3-dep
+
+# pointless lintian ...
+lib@PVER@-dbg binary: hardening-no-fortify-functions
+
+lib@PVER@-dbg binary: arch-dependent-file-not-in-arch-specific-directory
+
+# yes, some extensions don't have references to any external lib
+lib@PVER@-dbg binary: shared-lib-without-dependency-information
+lib@PVER@-dbg binary: library-not-linked-against-libc
+
+# pyexpat.c contains expat error messages that lintian mis-attributes to embedding
+lib@PVER@-dbg binary: embedded-library
--- /dev/null
+++ b/debian/libPVER-dbg.prerm.in
@@ -0,0 +1,23 @@
+#! /bin/sh
+
+set -e
+
+case "$1" in
+    remove)
+	dpkg -L lib@PVER@-dbg@HOST_QUAL@ \
+	    | awk '/\.py$/ {print $0"c\n" $0"o"}' \
+	    | xargs -r rm -f >&2
+        ;;
+    upgrade)
+        ;;
+    deconfigure)
+        ;;
+    failed-upgrade)
+        ;;
+    *)
+        echo "prerm called with unknown argument \`$1'" >&2
+        exit 1
+    ;;
+esac
+
+#DEBHELPER#
--- /dev/null
+++ b/debian/libPVER-dbg.symbols.in
@@ -0,0 +1,17 @@
+libpython@VER@d.so.1.0 libpython@VER@-dbg #MINVER#
+#include "libpython.symbols"
+ _PyInterpreterState_GetRefTotal@Base @SVER@
+ _PyObject_DebugMallocStats@Base @SVER@
+ _PySet_Dummy@Base @SVER@
+ _PyUnicode_CheckConsistency@Base @SVER@
+ _PyWideStringList_CheckConsistency@Base @SVER@
+ _Py_Dealloc@Base @SVER@
+ _Py_DECREF_DecRefTotal@Base @SVER@
+ _Py_GetGlobalRefTotal@Base @SVER@
+ _Py_GetLegacyRefTotal@Base @SVER@
+ _Py_HashSecret_Initialized@Base @SVER@
+ _Py_INCREF_IncRefTotal@Base @SVER@
+ _Py_NegativeRefcount@Base @SVER@
+ _Py_RefTotal@Base @SVER@
+ PyModule_Create2@Base @SVER@
+ PyModule_FromDefAndSpec2@Base @SVER@
--- /dev/null
+++ b/debian/libPVER.overrides.in
@@ -0,0 +1,4 @@
+lib@PVER@ binary: package-name-doesnt-match-sonames
+
+# pyexpat.c contains expat error messages that lintian mis-attributes to embedding
+lib@PVER@ binary: embedded-library
--- /dev/null
+++ b/debian/libPVER.symbols.in
@@ -0,0 +1,4 @@
+libpython@VER@.so.1.0 libpython@VER@ #MINVER#
+#include "libpython.symbols"
+ PyModule_Create2@Base @SVER@
+ PyModule_FromDefAndSpec2@Base @SVER@
--- a/debian/libPVERt64-dbg.overrides.in
+++ /dev/null
@@ -1,20 +0,0 @@
-lib@PVER@-dbg binary: package-name-doesnt-match-sonames
-lib@PVER@-dbg binary: non-dev-pkg-with-shlib-symlink
-
-# no, it's not unusual
-lib@PVER@-dbg binary: unusual-interpreter
-
-# just the gdb debug file
-lib@PVER@-dbg binary: python3-script-but-no-python3-dep
-
-# pointless lintian ...
-lib@PVER@-dbg binary: hardening-no-fortify-functions
-
-lib@PVER@-dbg binary: arch-dependent-file-not-in-arch-specific-directory
-
-# yes, some extensions don't have references to any external lib
-lib@PVER@-dbg binary: shared-lib-without-dependency-information
-lib@PVER@-dbg binary: library-not-linked-against-libc
-
-# pyexpat.c contains expat error messages that lintian mis-attributes to embedding
-lib@PVER@-dbg binary: embedded-library
--- a/debian/libPVERt64-dbg.prerm.in
+++ /dev/null
@@ -1,23 +0,0 @@
-#! /bin/sh
-
-set -e
-
-case "$1" in
-    remove)
-	dpkg -L lib@PVER@t64-dbg@HOST_QUAL@ \
-	    | awk '/\.py$/ {print $0"c\n" $0"o"}' \
-	    | xargs -r rm -f >&2
-        ;;
-    upgrade)
-        ;;
-    deconfigure)
-        ;;
-    failed-upgrade)
-        ;;
-    *)
-        echo "prerm called with unknown argument \`$1'" >&2
-        exit 1
-    ;;
-esac
-
-#DEBHELPER#
--- a/debian/libPVERt64-dbg.symbols.in
+++ /dev/null
@@ -1,17 +0,0 @@
-libpython@VER@d.so.1.0 libpython@VER@t64-dbg #MINVER#
-#include "libpython.symbols"
- _PyInterpreterState_GetRefTotal@Base @SVER@
- _PyObject_DebugMallocStats@Base @SVER@
- _PySet_Dummy@Base @SVER@
- _PyUnicode_CheckConsistency@Base @SVER@
- _PyWideStringList_CheckConsistency@Base @SVER@
- _Py_Dealloc@Base @SVER@
- _Py_DECREF_DecRefTotal@Base @SVER@
- _Py_GetGlobalRefTotal@Base @SVER@
- _Py_GetLegacyRefTotal@Base @SVER@
- _Py_HashSecret_Initialized@Base @SVER@
- _Py_INCREF_IncRefTotal@Base @SVER@
- _Py_NegativeRefcount@Base @SVER@
- _Py_RefTotal@Base @SVER@
- PyModule_Create2@Base @SVER@
- PyModule_FromDefAndSpec2@Base @SVER@
--- a/debian/libPVERt64.overrides.in
+++ /dev/null
@@ -1,4 +0,0 @@
-lib@PVER@ binary: package-name-doesnt-match-sonames
-
-# pyexpat.c contains expat error messages that lintian mis-attributes to embedding
-lib@PVER@ binary: embedded-library
--- a/debian/libPVERt64.symbols.in
+++ /dev/null
@@ -1,4 +0,0 @@
-libpython@VER@.so.1.0 libpython@VER@t64 #MINVER#
-#include "libpython.symbols"
- PyModule_Create2@Base @SVER@
- PyModule_FromDefAndSpec2@Base @SVER@
--- a/debian/libpython3.12t64-dbg.lintian-overrides
+++ /dev/null
@@ -1 +0,0 @@
-libpython3.12t64-dbg: package-name-doesnt-match-sonames libpython3.12-dbg
--- a/debian/libpython3.12t64.lintian-overrides
+++ /dev/null
@@ -1 +0,0 @@
-libpython3.12t64: package-name-doesnt-match-sonames libpython3.12
--- a/debian/rules
+++ b/debian/rules
@@ -279,7 +279,7 @@
 p_base	:= $(PVER)
 p_npie	:= $(PVER)-nopie
 p_min	:= $(PVER)-minimal
-p_lib	:= lib$(PVER)t64
+p_lib	:= lib$(PVER)
 p_gdbm	:= $(PVER)-gdbm
 p_tk	:= $(PVER)-tk
 p_dev	:= $(PVER)-dev
@@ -294,7 +294,7 @@
 p_lbase	:= lib$(PVER)-stdlib
 p_lmin	:= lib$(PVER)-minimal
 p_ldev	:= lib$(PVER)-dev
-p_ldbg	:= lib$(PVER)t64-dbg
+p_ldbg	:= lib$(PVER)-dbg
 p_ltst	:= lib$(PVER)-testsuite
 
 d_base	:= debian/$(p_base)
